voltdbclient/network_listener.go

```diff
@@ -0,0 +1,83 @@
+/* This file is part of VoltDB.
+ * Copyright (C) 2008-2016 VoltDB Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with VoltDB.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package voltdbclient
+
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+// NetworkListener listens for responses for asynchronous procedure calls from
+// the server.  If a callback (channel) is registered for the procedure, the
+// listener puts the response on the channel (calls back).
+type NetworkListener struct {
```

Comment:
Should this be called ProcedureResponseListener or something thats easy to relate to?